File: 01.05.一次编辑/01.05.go
func oneEditAway(first string, second string) bool {
    if len(first) > len(second) {
        first, second = second, first
    }
    if len(second) - len(first) > 1 {
        return false
    }

    for i := 0; i < len(first); i++ {
        if first[i] != second[i] {
            // 删除
            if first[i:] == second[i+1:] {
                return true
            }

            // 修改
            if first[i+1:] == second[i+1:] {
                return true
            }

            return false
        }
    }

    return true
}

File: 01.05.一次编辑/01.05.rs
impl Solution {
    pub fn one_edit_away(first: String, second: String) -> bool {
        let mut first_chars: Vec<char> = first.chars().collect();
        let mut second_chars: Vec<char> = second.chars().collect();

        if first_chars.len() > second_chars.len() {
            let mut tmp: Vec<char> = first_chars;
            first_chars = second_chars;
            second_chars = tmp;
        }
        if second_chars.len() - first_chars.len() > 1 {
            return false;
        }

        let mut i = 0;
        while i < first_chars.len() {
            if first_chars[i] != second_chars[i] {
                if first_chars[i..] == second_chars[i + 1..] {
                    return true;
                }
                if first_chars[i + 1..] == second_chars[i + 1..] {
                    return true;
                }
                return false;
            }
            i += 1;
        }
        true
    }
}
